### PR TITLE
[ENG-965] Perf improvement

### PR DESCRIPTION
**Pull request by heathdutton**
These query changes allow indexes to be used and prevent blocking transactions, especially for scheduled client maintenance.

### CODE REVIEW

File: Model/Cache.php [hunk 1]
```diff
@@ -168,6 +172,29 @@ private function getContainer()
         return $this->container;
     }
 
+    /**
+     * Shore up EntityManager loading, in case there is a flaw in a plugin or campaign handling.
+     *
+     * @return EntityManager
+     */
+    private function getEntityManager()
```
Comment (dean-martin) [inline Model/Cache.php]: Maybe consider making this a trait? This is used frequently.. Could replace 'ContactClient' with get_class($this)

File: Model/FilePayload.php [hunk 2]
```diff
@@ -238,11 +247,37 @@ public function setAttribution($attribution)
...
+    private function getEntityManager()
+    {
+        try {
+            if ($this->em && !$this->em->isOpen()) {
+                $this->em = $this->em->create(
+                    $this->em->getConnection(),
+                    $this->em->getConfiguration(),
+                    $this->em->getEventManager()
+                );
+                // $this->logger->error('ContactClient: EntityManager was closed.');
```
Comment (dean-martin) [inline Model/FilePayload.php]: Should this be uncommented as well?

Also looks like this should be a trait, or some kind of service perhaps....

File: Model/Schedule.php [hunk 3]
```diff
@@ -357,11 +362,37 @@ private function evaluateFileRate($fileRate = 1, $date = null)
...
+    private function getEntityManager()
+    {
+        try {
+            if ($this->em && !$this->em->isOpen()) {
+                $this->em = $this->em->create(
+                    $this->em->getConnection(),
+                    $this->em->getConfiguration(),
+                    $this->em->getEventManager()
+                );
+                // $this->logger->error('ContactClient: EntityManager was closed.');
```
Comment (dean-martin) [inline Model/Schedule.php]: Commented out here too

File: Event/ContactClientStatEvent.php [hunk 4]
```diff
@@ -91,10 +92,25 @@ public function getContact()
     }
 
     /**
+     * Shore up EntityManager loading, in case there is a flaw in a plugin or campaign handling.
+     *
      * @return EntityManager
      */
-    public function getEntityManager()
+    private function getEntityManager()
```
[the remark this answers is not in the document]
Comment (heathdutton) [reply Event/ContactClientStatEvent.php]: Hmmm.

File: Model/FilePayload.php (hunk 2 above)
Comment (heathdutton) [reply Model/FilePayload.php]: I should have made this a trait, but a trait would then need the logger service... I'm hoping that we don't actually need any of these getEntityManager() methods and we can drop them in the next release. I'm just trying to stop some fires.

File: Model/Schedule.php (hunk 3 above)
Comment (heathdutton) [reply Model/Schedule.php]: Logger isn't available here... Again, this is a bandaid. :(

File: Helper/TokenHelper.php [hunk 5]
```diff
@@ -548,7 +552,7 @@ private function addHelper($type)
                                             // 12223334444
                                             $value = substr($phone, 1);
```
[the remark this answers is not in the document]
Comment (heathdutton) [reply Helper/TokenHelper.php]: Really? for trimming off the first char? Seems overkill. The phoneHelper has already done the work here.

**Comment by heathdutton** [comment]
Conflicts resolved.

**Pull request merged**